test/unit/parse.cpp

```diff
@@ -368,6 +368,14 @@ TEST_CASE("parse registry entry", "[parse]") {
             REQUIRE(r);
         }
     }
+
+    // invalid input that was involved in a crash
+    {
+        auto r = uenv::parse_registry_entry(
+            "service/eiger/zen2/prgenv/test/v1/1902620939");
+        REQUIRE(r->nspace == "service");
+        REQUIRE(!r);
```

Comment:
`REQUIRE(r->nspace == "service")` is an invalid call I think.
It should only be
```
REQUIRE(!r)
```